### ISSUE TITLE
PayPal Commerce is NOT pickup up the Payment Address name for credit cards

### ISSUE THREAD

**Issue by goatlady102** [abantecart/abantecart-src#1554]
In testing PayPal Commerce v 1.00 default extension in AbanteCart v 1.3.3 I am encountering an issue,  that it does NOT pickup the Payment address so that a credit card can be processed.

It pickups up the Billing Address.   I cannot find any settings that allow me to redirect the extension to use the Payment address.  Therefore I assume the extension is in error.

![Paypal Payment addres](https://user-images.githubusercontent.com/7943102/197285422-b852ecae-16f3-4ed4-b957-b48e3cea644b.png)

**Comment by BasaraABC** [comment]
Hello. in your screenshot PayPal form use the Payment address you entered in checkout which is correct. However, if the address did not pass internal PayPal's address validation the form to enter the address will appear. Please see an example with the accepted address on the left and the same address with the incorrect zip on the right.
![invalid](https://user-images.githubusercontent.com/2631910/197458442-453fc669-77fd-494d-9216-50a7addb64fc.png)

**Comment by goatlady102** [comment]
i believe you missed my point -  the PAYMENT info in the testing site  has a different name than the account.  However  the PayPal  credit card form ignores the PAYMENT info  and uses the account info.   The reason this is important is customers often use there nickname for the their account info,  in my case it is Lee.   However my cards are issues in my full legal name or my spouse, etc.  The form  needs to use the info as address in the PAYMENT/Billing  part of the account information.

**Comment by BasaraABC** [comment]
@abolabo to replicate the issue set up two addresses in the registered account:
Default address1 with name1 and additional address with name2. At checkout, the customer set the billing address to the second one but PayPal takes the name from address 1. See screenshot
<img width="1286" alt="billing" src="https://user-images.githubusercontent.com/2631910/197687855-3ded07db-f0b3-4257-80d9-bc271f20b89f.png">

**Comment by goatlady102** [comment]
Hi again,
This issue is really bothering me -  is there any way you can display the Payment info that is going to be used before the customers makes the submittal to PayPal?  Adding a note that says the payment info needs to match what the cart account has would be of great benefit, too. Or can you make it so the form with the failure message can be updated and resubmitted to PayPal.

**Comment by BasaraABC** [comment]
There is no way to do this. The buttons are loaded completely by 3rd party (PayPal) and we have no control over it.

**Comment by goatlady102** [comment]
question - today  10/31  I was retesting PayPal  in preparation to debug.   However,  there is no issue with the card payment,  and there is NO validation of the name or address.   I tested using  my name for account and using my spouses name,  different card address,  and did NOT change the payment address -  and PayPal processed without an issue.  

So my question is why?  did you have the ability to change your Abantecart connections with PayPal to not validate?
I made no changes in my testing account from the previous testing session,  when things failed.

thanks for the feedback, I need to understand what/where something changed before I update my managed clients.
